features/conversation-list/use-conversation-list-conversations.tsx

```diff
@@ -0,0 +1,45 @@
+import { useCurrentAccount } from "@/data/store/accountsStore";
+import { isConversationAllowed } from "@/features/conversation/utils/is-conversation-allowed";
+import { getConversationMetadataQueryOptions } from "@/queries/conversation-metadata-query";
+import { getConversationsQueryOptions } from "@/queries/conversations-query";
+import { useQueries, useQuery } from "@tanstack/react-query";
+import { useMemo } from "react";
+
+export const useConversationListConversations = () => {
+  const currentAccount = useCurrentAccount();
+
+  const { data: conversations, ...rest } = useQuery(
+    getConversationsQueryOptions({
+      account: currentAccount!,
+    })
+  );
+
+  const conversationsDataQueries = useQueries({
+    queries: (conversations ?? []).map((conversation) =>
+      getConversationMetadataQueryOptions({
+        account: currentAccount!,
+        topic: conversation.topic,
+      })
+    ),
+  });
+
+  const conversationsFiltered = useMemo(() => {
```

Comment:
I believe we can use `combine` for this: 

![CleanShot 2025-01-16 at 16.01.01@2x.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/PNcXAJ13O5q4zOLQxaRk/8ceb6b18-3cc5-46b6-832c-b8c32f3e8ccf.png)

Motivation for doing so is using an array in the dependency array generally blows up in my experience due to reference vs value comparison